proposal.html

```diff
@@ -485,7 +485,7 @@
 
   <ins class="block">
     <emu-alg>
-      1. If _func_ is a <emu-xref href="#sec-bound-function-exotic-objects">Bound Function exotic object</emu-xref> or a <emu-xref href="#sec-ecmascript-standard-built-in-objects">built-in Function object</emu-xref>, then return an implementation-dependent String source code representation of _func_. The representation must have the syntax of a |NativeFunction|.
+      1. If _func_ is a <emu-xref href="#sec-bound-function-exotic-objects">Bound Function exotic object</emu-xref> or a <emu-xref href="#sec-ecmascript-standard-built-in-objects">built-in Function object</emu-xref>, then return an implementation-dependent String source code representation of _func_. The representation must have the syntax of a |NativeFunction|. Additionally, if _func_ is a <emu-xref href="#sec-well-known-intrinsic-objects">Well-known Intrinsic Object</emu-xref> and the Global Name associated with _func_ has the syntax of a |BindingIdentifier|, the portion of the returned String that would be matched by |BindingIdentifier| must be the same as the Global Name associated with _func_.
```

Comment:
So would such a `Math.sin` function not be a "built-in Function object"? And if it isn't, wouldn't it get caught by step 2 ("_func_ has a \[[SourceText]] internal slot and _func_.[[SourceText]] is not *undefined*") before ever making it to step 3? What callable objects are not "function-typed built-in" or bound, but _don't_ have \[[SourceText]]?

In short, why is there a \[[SourceText]] check in between two _NativeFunction_ cases?